nntrainer/tensor/cache_pool.cpp

```diff
@@ -11,51 +11,81 @@
  *
  */
 
-#include "memory_pool.h"
+#include "cache_pool.h"
+
 #include <limits>
 #include <numeric>
 #include <stdexcept>
 #include <vector>
 
-#include <cache_pool.h>
 #include <nntrainer_error.h>
 #include <nntrainer_log.h>
 #include <profiler.h>
 
 namespace nntrainer {
 
-void CacheElem::swapIn() {
-  std::lock_guard<std::mutex> lock(device_mutex);
-  void *buf = device->getBuffer(offset, length);
-  mem_data->setAddr((float *)buf);
-  mem_data->setValid(true);
-  active = true;
+namespace {
+
+inline const CachePolicy
+convertTensorLifespanToCachePolicy(const TensorLifespan lifespan) {
+  CachePolicy policy;
+
+  switch (lifespan) {
+  case TensorLifespan::UNMANAGED:
+    policy = CachePolicy::ALWAYS_SYNCED;
+    break;
+  case TensorLifespan::FORWARD_FUNC_LIFESPAN:
+    policy = CachePolicy::ALWAYS_SYNCED;
```

Comment:
I'm not following why ```FORWARD_FUNC_LIFESPAN``` is needs to be ```ALWAYS_SYNCED```?

Reply:
Known var/grad tensors donot have this lifespan. To avoid error for unconsidered tensors, it sets ALWAYS_SYNCED for this.

Reply:
Well, for the var tensor for non-trainable layer, we could have FORWARD_FUNC_LIFESPAN and we do not need to be ALWAYS_SYNCED.

Reply:
Current implementation for non-trainable does not distinguished by lifespan, it's excluding execution order when checking  lifespan.

Reply:
It changed to TEMPORAL as it is.